test/ng/httpSpec.js

```diff
@@ -1369,17 +1369,15 @@ describe('$http', function() {
             }
           );
 
-          it('should forward json deserialization errors to the http error handler',
-              function() {
+          it('should return JSON data with error message if JSON is invalid', function() {
             var errCallback = jasmine.createSpy('error');
-
-            $httpBackend.expect('GET', '/url').respond('abcd', {'Content-Type': 'application/json'});
-            $http({method: 'GET', url: '/url'}).then(callback).catch(errCallback);
+            $httpBackend.expect('GET', '/url').respond('{abcd}', {'Content-Type': 'application/json'});
```

Comment:
@chirag64, yes, I am referring to the string. Why change `abcd` to `{abcd}`?
<sub>I am mostly curious - I don't think it makes any difference.</sub>